universum/analyzers/uncrustify.py

```diff
@@ -1,200 +1,202 @@
-        if not files:
-            sys.stderr.write("Please provide at least one file for analysis")
-            return 2
-
-        return files
-
-    def get_htmldiff_parameters(self):
-        if not (self.wrapcolumn and self.tabsize):
-            with open(self.settings.cfg_file) as config:
-                for line in config.readlines():
-                    if line.startswith("code_width"):
-                        self.wrapcolumn = int(line.split()[2])
-                    if line.startswith("input_tab_size"):
-                        self.tabsize = int(line.split()[2])
-        return self.wrapcolumn, self.tabsize
-
-    def generate_html_diff(self, file_name, left_lines, right_lines):
-        wrapcolumn, tabsize = self.get_htmldiff_parameters()
-        differ = difflib.HtmlDiff(tabsize=tabsize, wrapcolumn=wrapcolumn)
-        file_name = os.path.relpath(file_name, self.settings.output_directory).replace('/', '_') + '.html'
-        with open(os.path.join(self.settings.output_directory, file_name), 'w') as outfile:
-            outfile.write(differ.make_file(left_lines, right_lines, context=False))
-
-    def get_file_issues(self, src_file):
-        self.settings.output_directory = os.path.join(os.getcwd(), self.settings.output_directory)
-        # Uncrustify copies absolute path in its target folder, that's why we use '+'
-        uncrustify_file = os.path.normpath(self.settings.output_directory + '/' + src_file)
-
+def _uncrustify_output_parser(files: List[Tuple[Path, Path]],
+                              diff_file_writer: HtmlDiffFileWriter,
+                              _: str) -> List[utils.ReportData]:
+    result: List[utils.ReportData] = []
+    for src_file, uncrustify_file in files:
         with open(src_file) as src:
             src_lines = src.readlines()
         with open(uncrustify_file) as fixed:
             fixed_lines = fixed.readlines()
 
-        file_issues = []
-        matching_blocks = difflib.SequenceMatcher(a=src_lines, b=fixed_lines).get_matching_blocks()
-        previous_match = matching_blocks[0]
-        for match in matching_blocks[1:]:
-            block = get_mismatching_block(previous_match, match, src_lines, fixed_lines)
-            previous_match = match
-            if block:
-                file_issues.append(get_issue_json_format(src_file, block))
-
-        # Generate html diff
-        if file_issues:
-            self.generate_html_diff(uncrustify_file, src_lines, fixed_lines)
-
-        return file_issues
-
-    def execute(self):
-        if not self.settings.cfg_file and ('UNCRUSTIFY_CONFIG' not in os.environ):
-            sys.stderr.write("Please specify the '--cfg_file' parameter "
-                             "or set an env. variable 'UNCRUSTIFY_CONFIG'")
-            return 2
+        issues = _get_issues_from_diff(src_file, src_lines, fixed_lines)
+        if issues:
+            diff_file_writer.write_file(src_file, src_lines, fixed_lines)
```

Comment:
writing html diff file used to be a part of get_file_issues function, but it seems to me that dependency on html is not needed here. In my opinion, parameter data such as target directory and format details for diff files does not belong to this function. Moved to separate 'writer' class, which is built ahead of time in main().